### PR TITLE
Delete low-value PAL test

### LINKED ISSUE #48496: Test failed : threading.SetThreadDescription.test1
Run: [runtime-coreclr outerloop 20210218.4](https://dev.azure.com/dnceng/public/_build/results?buildId=1001621&view=ms.vss-test-web.build-test-results-tab&runId=31316134&resultId=109694&paneView=debug)

Failed test:
```
CoreCLR Linux x64 Checked no_tiered_compilation @ (Alpine.312.Amd64.Open)ubuntu.1604.amd64.open@mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.12-helix-20200602002622-e06dc59

 -threading.SetThreadDescription.test1
```

**Error message:**
```
paltests <PrintPalTests|TestName>
Either print list of all paltests by passing PrintPalTests, or run a single PAL test.
```

### PR DESCRIPTION
**Pull request by jkotas**
This partially reverts commit 77c939c9c9acfc49fcb7dc695ff33d03ca06f99e.

Fixes #48496

### CODE REVIEW

**Comment by danmoseley** [comment]
Without this test, couldn't I replace the body of `SetThreadDescription(..)` in the PAL with `return 0;` and all our tests would pass? This is the only test that verifies we actually set the native thread name.

Why not keep this test, which is working fine, and make the fix suggested in #48496?

File: src/coreclr/pal/tests/palsuite/compilableTests.txt [hunk 1]
```diff
@@ -772,13 +772,11 @@ threading/ResetEvent/test2/paltest_resetevent_test2
 threading/ResetEvent/test3/paltest_resetevent_test3
 threading/ResetEvent/test4/paltest_resetevent_test4
 threading/ResumeThread/test1/paltest_resumethread_test1
-threading/SwitchToThread/test1/paltest_switchtothread_test1
```
Comment (BruceForstall) [inline src/coreclr/pal/tests/palsuite/compilableTests.txt:775]: Is this related?

**Review by BruceForstall — APPROVED** [review]
I'm ok with this if you think it's the right thing to do.

**Comment by jkotas** [comment]
> Without this test, couldn't I replace the body of SetThreadDescription(..) in the PAL with return 0; and all our tests would pass?

Yep. There are many other ways how one can break the connection between `Thread.Name` managed API and the OS name that this test is not catching either. It is what makes this test low-value.

If we wanted to have actual testing here, we would be testing that setting `Thread.Name` is actually going to show up in the diagnostic tools.

> Why not keep this test, which is working fine, and make the fix suggested in #48496?

It is not worth my time to work on tweaking this to make it pass. If you would like to work on this, feel free to close this PR and submit alternative.

**Comment by danmoseley** [comment]
Fair enough

**Comment by jkotas** [comment]
Also, setting of the OS thread name would be ideally shared between runtimes, via the shared CoreLib and System.Native shim.

File: src/coreclr/pal/tests/palsuite/compilableTests.txt (hunk 1 above)
Comment (jkotas) [reply src/coreclr/pal/tests/palsuite/compilableTests.txt:775]: This line is redundant - the same line is ~10 lines below already. I think this was just an incorrectly resolved conflict in the original PR.
